src/Context/SoapContext.php

```diff
@@ -215,4 +237,18 @@ public function savedValueNotMatchesRegExp($pattern)
     {
         Assertions::assertNotRegExp($pattern, $this->value);
     }
+
+    /**
+     * @BeforeStep
+     */
+    public function beforeStep(BeforeStepScope $scope)
```

Comment:
Can we rename the method to explicitly say that we're working with exceptions? Like `beforeStepCheckException`?